oarepo_ui/templating/catalog.py

```diff
@@ -0,0 +1,459 @@
+#
+# Copyright (c) 2025 CESNET z.s.p.o.
+#
+# This file is a part of oarepo-ui (see https://github.com/oarepo/oarepo-ui).
+#
+# oarepo-ui is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+#
+"""OARepo UI templating catalog module.
+
+This module provides an extended Jinjax catalog implementation for OARepo UI,
+including component rendering with asset management, template loading,
+and integration with Flask's static file serving for CSS and JavaScript assets.
+"""
+
+from __future__ import annotations
+
+import os
+import re
+from itertools import chain
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, NamedTuple, cast, override
+
+import flask
+import jinja2
+from flask import current_app
+from flask.globals import request
+from jinjax import Catalog
+from jinjax.exceptions import ComponentNotFound
+from jinjax.jinjax import JinjaX
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+
+    from flask import Flask
+    from jinjax import Component
+
+DEFAULT_URL_ROOT = "/static/components/"
+ALLOWED_EXTENSIONS = (".css", ".js")
+DEFAULT_PREFIX = ""
+DEFAULT_EXTENSION = ".jinja"
+DELIMITER = "."
+SLASH = "/"
+PROP_ATTRS = "attrs"
+PROP_CONTENT = "content"
+
+
+class SearchPathItem(NamedTuple):
+    """Named tuple for search path items in the Jinjax catalog."""
+
+    template_name: str
+    absolute_path: Path
+    relative_path: Path
+    priority: int
+
+
+class OarepoCatalog(Catalog):
```

Comment:
Do we need to override so many things here? Many methods look very similar to: https://github.com/jpsca/jinjax/blob/main/src/jinjax/catalog.py